packages/@react-aria/autocomplete/src/useAutocomplete.ts

```diff
@@ -0,0 +1,244 @@
+  // for screen reader announcements
+  let lastInputValue = useRef<string | null>(null);
+  useEffect(() => {
+    if (state.inputValue != null) {
+      if (lastInputValue.current != null && lastInputValue.current !== state.inputValue && lastInputValue.current?.length <= state.inputValue.length) {
+        focusFirstItem();
+      } else {
+        clearVirtualFocus();
+      }
+
+      lastInputValue.current = state.inputValue;
+    }
+  }, [state.inputValue, focusFirstItem, clearVirtualFocus]);
+
+  // For textfield specific keydown operations
+  let onKeyDown = (e: BaseEvent<ReactKeyboardEvent<any>>) => {
+    if (e.nativeEvent.isComposing) {
+      return;
+    }
+
+    switch (e.key) {
+      case 'Escape':
+        // Early return for Escape here so it doesn't leak the Escape event from the simulated collection event below and
+        // close the dialog prematurely. Ideally that should be up to the discretion of the input element hence the check
+        // for isPropagationStopped
+        if (e.isPropagationStopped()) {
+          return;
+        }
+        break;
+      case ' ':
+        // Space shouldn't trigger onAction so early return.
+
+        return;
+      case 'Home':
+      case 'End':
+      case 'PageDown':
+      case 'PageUp':
+      case 'ArrowUp':
+      case 'ArrowDown': {
+        // Prevent these keys from moving the text cursor in the input
+        e.preventDefault();
+        // Move virtual focus into the wrapped collection
+        let focusCollection = new CustomEvent(FOCUS_EVENT, {
+          cancelable: true,
+          bubbles: true
+        });
+
+        collectionRef.current?.dispatchEvent(focusCollection);
+        break;
+      }
+      case 'ArrowLeft':
+      case 'ArrowRight':
+        // TODO: will need to special case this so it doesn't clear the focused key if we are currently
+        // focused on a submenutrigger? May not need to since focus would
+        // But what about wrapped grids where ArrowLeft and ArrowRight should navigate left/right
+        clearVirtualFocus();
+        break;
+    }
+
+    // Emulate the keyboard events that happen in the input field in the wrapped collection. This is for triggering things like onAction via Enter
+    // or moving focus from one item to another
+    if (state.focusedNodeId == null) {
+      collectionRef.current?.dispatchEvent(
+        new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
+      );
+    } else {
+      let item = collectionRef.current?.querySelector(`#${CSS.escape(state.focusedNodeId)}`);
+      item?.dispatchEvent(
+        new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
+      );
+      // TODO: this currently has problems making Enter trigger Listbox links. usePress catches the press up that happens but
+      // detects that the press target is different from the event target aka listbox item vs the input where the Enter event occurs...
```

Comment:
I had thought that was the reason when I ran into this yesterday, but it didn't seem to change anything, the keyup event in usePress still detects that the event target of the keyup is the input. I assumed that stopping propagation on keydown would've allowed the dispatched keyup event to make it to usePress and fake that the keyup was happening on the option rather than the input but no luck

Reply:
I guess because the key up event on the input happens first. Maybe capturing listener would work?

Reply:
unfortunately making the onKeyUp a capturing listener didn't seem to help, the event on the input still happened first it seems. Changing the capturing listener for onKeyUp in usePress to non-capturing did fix it but we rely on it being capturing...

Reply:
took another stab at this and managed to fix it by attaching the capturing listener to the document so it can get called before usePress's capturing listener